### PR TITLE
fix(storefront): BCTHEME-906 Hide prices for aria-label and data-product-price attributes if set to "Hidden for guests". Hide currency selection for non-logged in users.

### PR DESCRIPTION
**Pull request by bc-yevhenii-buliuk**
#### What?
This PR hides prices for non-logged in users in the aria-label and data-product-price attributes.
This PR hides currency selection for non-logged in users.
Also this PR fixes translation gap in Cornerstone for "Log in for pricing" phrase.

#### Tickets / Documentation
- [BCTHEME-906](https://jira.bigcommerce.com/browse/BCTHEME-906)

#### Screenshots
906_price_in_aria-label_and_data-product-price
906_hidden_price

### CODE REVIEW

File: templates/components/products/card.html [hunk 1]
```diff
@@ -17,11 +17,15 @@
         {{/each}}"
         data-product-brand="{{brand.name}}"
         data-product-price="
-        {{#if price.with_tax}}
-            {{price.with_tax.value}}
+        {{#or customer (if theme_settings.restrict_to_login '!==' true)}}
```
Comment (yurytut1993) [inline templates/components/products/card.html:20]: Can we avoid inversion in if statement?

File: templates/components/products/card.html [hunk 2]
```diff
@@ -17,11 +17,15 @@
...
+            {{#if price.with_tax}}
+                {{price.with_tax.value}}
+            {{else}}
+                {{price.without_tax.value}}
+            {{/if}}
         {{else}}
-            {{price.without_tax.value}}
-        {{/if}}"
+            Log in for pricing
```
Comment (yurytut1993) [inline templates/components/products/card.html:27]: Should we add translations here?

File: templates/components/products/card.html [hunk 3]
```diff
@@ -39,9 +43,15 @@
                 }}
             {{/or}}
         {{/if}}
+        
         <a href="{{url}}"
            class="card-figure__link"
-           aria-label="{{name}},{{> components/products/product-aria-label}}"
+           aria-label="
+           {{#or customer (if theme_settings.restrict_to_login '!==' true)}}
```
Comment (yurytut1993) [inline templates/components/products/card.html:50]: `theme_settings.restrict_to_login '===' false)` ?

File: templates/components/products/card.html (hunk 1 above)
Comment (bc-yevhenii-buliuk) [reply templates/components/products/card.html]: @yurytut1993 done!

File: templates/components/products/card.html (hunk 3 above)
Comment (bc-yevhenii-buliuk) [reply templates/components/products/card.html]: @yurytut1993 done!

File: templates/components/products/list-item.html [hunk 4]
```diff
@@ -1,12 +1,26 @@
...
-    <article class="listItem" data-event-type="{{event}}" data-entity-id="{{id}}" data-position="{{position}}" data-name="{{name}}" data-product-category="{{#each category}}{{#if @last}}{{this}}{{else}}{{this}}, {{/if}}{{/each}}" data-product-brand="{{brand.name}}" data-product-price="{{#if price.with_tax}}{{price.with_tax.value}}{{else}}{{price.without_tax.value}}{{/if}}">
+    <article class="listItem" data-event-type="{{event}}" data-entity-id="{{id}}" data-position="{{position}}" data-name="{{name}}" data-product-category="{{#each category}}{{#if @last}}{{this}}{{else}}{{this}}, {{/if}}{{/each}}" data-product-brand="{{brand.name}}" data-product-price="
+    {{#or customer (if theme_settings.restrict_to_login '===' false)}}
+        {{#if price.with_tax}}
+            {{price.with_tax.value}}
+        {{else}}
+            {{price.without_tax.value}}
+        {{/if}}
+    {{else}}
+        Log in for pricing
```
Comment (bc-yevhenii-buliuk) [inline templates/components/products/list-item.html:10]: @yurytut1993 Sorry, I accidentally deleted your comment about the translation of this phrase, but I checked the lang.json files and there are no translations for it.

Comment (yurytut1993) [reply templates/components/products/list-item.html:10]: can we use `{{> components/common/login-for-pricing}}` instead of simple text?
also looks like we have translation gap in Cornerstone

File: templates/components/products/card.html [hunk 5]
```diff
@@ -103,7 +113,12 @@
             <p class="card-text" data-test-info-type="brandName">{{brand.name}}</p>
         {{/if}}
         <h3 class="card-title">
-            <a aria-label="{{name}},{{> components/products/product-aria-label}}"
+            <a aria-label="
+               {{#or customer (if theme_settings.restrict_to_login '===' false)}}
+                   {{name}},{{> components/products/product-aria-label}}
+               {{else}}
+                   {{name}}
+               {{/or}}"
```
[the remark this answers is not in the document]
Comment (bc-yevhenii-buliuk) [reply templates/components/products/card.html]: @bc-alexsaiannyi yes, I have moved this into a separate component.

File: templates/components/products/card.html [hunk 6]
```diff
@@ -39,9 +43,15 @@
                 }}
             {{/or}}
         {{/if}}
+        
         <a href="{{url}}"
            class="card-figure__link"
-           aria-label="{{name}},{{> components/products/product-aria-label}}"
+           aria-label="
+           {{#or customer (if theme_settings.restrict_to_login '===' false)}}
```
[the remark this answers is not in the document]
Comment (bc-yevhenii-buliuk) [reply templates/components/products/card.html]: @bc-alexsaiannyi I think we should leave the condition **#or** because we need to test two conditionals, not just a negative conditional.

[the remark this answers is not in the document]
Comment (bc-alexsaiannyi) [reply templates/components/products/card.html]: inside  `or` helper you're checking `theme_settings.restrict_to_login` for falsy value. And  `unless` helper can be used  as the inverse of the `if` helper what means that Its block will be rendered if the expression returns a falsy value.

[the remark this answers is not in the document]
Comment (bc-yevhenii-buliuk) [reply templates/components/products/card.html]: @bc-alexsaiannyi done!

File: templates/components/products/product-view.html [hunk 7]
```diff
@@ -1,4 +1,15 @@
-<div class="productView" {{#if settings.data_tag_enabled}} data-event-type="product" data-entity-id="{{product.id}}" data-name="{{product.title}}" data-product-category="{{#each product.category}}{{#if @last}}{{this}}{{else}}{{this}}, {{/if}}{{/each}}" data-product-brand="{{product.brand.name}}" data-product-price="{{#if product.price.with_tax}}{{product.price.with_tax.value}}{{else}}{{product.price.without_tax.value}}{{/if}}" data-product-variant="single-product-option"{{/if}}>
+<div class="productView" {{#if settings.data_tag_enabled}} data-event-type="product" data-entity-id="{{product.id}}" data-name="{{product.title}}" data-product-category="{{#each product.category}}{{#if @last}}{{this}}{{else}}{{this}}, {{/if}}{{/each}}" data-product-brand="{{product.brand.name}}"
```
Comment (bc-alexsaiannyi) [inline templates/components/products/product-view.html:1]: it's not related to your code but can you  also format this block of handlebars code so it would be easier to read. In a manner as it's done with data-product-price attribute.

**Review by bc-vlad-dlogush — APPROVED** [review]

**Review by yurytut1993 — APPROVED** [review]
